### PR TITLE
Add zipf distributing ratelimiter

### PR DESCRIPTION
**Pull request by oschaaf**
Adds a probabilistic filtering rate limiter, which will suppress an underlying rate limiter's pacing using a zipf distribution.

Signed-off-by: Otto van der Schaaf <oschaaf@we-amp.com>

### CODE REVIEW

**Comment by oschaaf** [comment]
Note to self: address first round of comment from #218 here before proceeding with review

File: source/common/rate_limiter_impl.h [hunk 1]
```diff
@@ -203,4 +204,30 @@ class GraduallyOpeningRateLimiterFilter : public FilteringRateLimiterImpl {
...
+public:
+  /**
+   * From the absl header associated to the zipf distribution:
+   * Preconditions: v > 0, q > 1
+   * The precondidtions are validated when NDEBUG is not defined via
+   * a pair of assert() directives.
+   * If NDEBUG is defined and either or both of these parameters take invalid
+   * values, the behavior of the class is undefined.
+   */
+  ZipfRateLimiterImpl(RateLimiterPtr&& rate_limiter, bool deterministic, double q = 2.0,
```
Comment (mum4k) [inline source/common/rate_limiter_impl.h]: I believe we had some open comments for the API and its documentation here in the original PR where this was introduced before we split it out. Would you mind resolving those comments before we proceed?

Comment (oschaaf) [reply source/common/rate_limiter_impl.h]: Done, addressed the [prior feedback round](https://github.com/envoyproxy/nighthawk/pull/218#pullrequestreview-335035033) in 10dbc29f4affec32379b40a4e0d9a3f117a36474

**Comment by mum4k** [comment]
Could you update the description of this PR? Unless there is still something blocking this?

**Comment by oschaaf** [comment]
> Could you update the description of this PR? Unless there is still something blocking this?

Done!

**Pull request merged**